### PR TITLE
Parse DS records

### PR DESCRIPTION
**Pull request by kmkaplan**
Adds support for parsing DS records from zone files.

This requires a "breaking" change: the TTL parser used to accept raw unit letters as valid values. Thus `DS` was parsed as a 0 second TTL. This PR makes that invalid,

### CODE REVIEW

File: crates/client/src/serialize/txt/parse_rdata.rs [hunk 1]
```diff
@@ -69,7 +71,10 @@ impl RDataParser for RData {
                 return Err(ParseError::from("CDNSKEY should be dynamically generated"))
             }
             RecordType::KEY => return Err(ParseError::from("KEY should be dynamically generated")),
-            RecordType::DS => return Err(ParseError::from("DS should be dynamically generated")),
+            #[cfg(feature = "dnssec")]
+            RecordType::DS => RData::DNSSEC(DNSSECRData::DS(ds::parse(tokens)?)),
+            #[cfg(not(feature = "dnssec"))]
+            RecordType::CDS => return Err(ParseError::from("DS should be dynamically generated")),
```
Comment (bluejekyll) [inline crates/client/src/serialize/txt/parse_rdata.rs:77]: I think this cfg flag in the not case is meant to be DS, right?

**Review by bluejekyll — CHANGES_REQUESTED** [review]
I think overall this looks good. It looks like maybe there's a copy-paste typo.

File: crates/client/src/serialize/txt/parse_rdata.rs (hunk 1 above)
Comment (kmkaplan) [reply crates/client/src/serialize/txt/parse_rdata.rs]: Right.
I have fixed that plus the other failures.

**Comment by bluejekyll** [comment]
Looks great, thanks for the PR!

**Pull request merged**